[engine] TestExampleMaster

### Which jobs are flaking?

TestExampleMaster

### Which test(s) are flaking?

engine-UT

### Jenkins logs or GitHub Actions link

```log
[2022-08-08T15:32:44.451Z] [2022/08/08 23:31:47.865 +08:00] [ERROR] [callbacks.go:134] ["trace log"] [component=gorm] [elapsed=705.366µs] [sql="UPDATE `master_meta_kv_data` SET `address`=\"\",`config`=\"\",`epoch`=0,`id`=\"master\",`node_id`=\"\",`project_id`=\"\",`status`=\"2\",`type`=\"0\",`updated_at`=\"2022-08-08 23:31:47.864\" WHERE id = \"master\" AND `master_meta_kv_data`.`deleted` IS NULL"] [affected-rows=0] [error="database table is locked"]

[2022-08-08T15:32:44.451Z] [2022/08/08 23:31:47.865 +08:00] [DEBUG] [callbacks.go:134] ["trace log"] [component=gorm] [elapsed=465.267µs] [sql="SELECT * FROM `logic_epoches` WHERE job_id = \"\" ORDER BY `logic_epoches`.`seq_id` LIMIT 1"] [affected-rows=1]

[2022-08-08T15:32:44.451Z]     master_impl_test.go:75: 

[2022-08-08T15:32:44.451Z]         	Error Trace:	/home/jenkins/agent/workspace/ut-check/tiflow/engine/jobmaster/example/master_impl_test.go:75

[2022-08-08T15:32:44.451Z]         	Error:      	Received unexpected error:

[2022-08-08T15:32:44.451Z]         	            	[DFLOW:ErrMetaOpFail]meta operation fail: database table is locked

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/errors.AddStack

[2022-08-08T15:32:44.451Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/errors.Trace

[2022-08-08T15:32:44.451Z]         	            		/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/tiflow/engine/framework/metadata.(*MasterMetadataClient).Update

[2022-08-08T15:32:44.451Z]         	            		/home/jenkins/agent/workspace/ut-check/tiflow/engine/framework/metadata/metadata.go:74

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).markStatusCodeInMetadata

[2022-08-08T15:32:44.451Z]         	            		/home/jenkins/agent/workspace/ut-check/tiflow/engine/framework/master.go:533

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).Init

[2022-08-08T15:32:44.451Z]         	            		/home/jenkins/agent/workspace/ut-check/tiflow/engine/framework/master.go:308

[2022-08-08T15:32:44.451Z]         	            	github.com/pingcap/tiflow/engine/jobmaster/example.TestExampleMaster

[2022-08-08T15:32:44.451Z]         	            		/home/jenkins/agent/workspace/ut-check/tiflow/engine/jobmaster/example/master_impl_test.go:74

[2022-08-08T15:32:44.451Z]         	            	testing.tRunner

[2022-08-08T15:32:44.451Z]         	            		/usr/local/go/src/testing/testing.go:1439

[2022-08-08T15:32:44.451Z]         	            	runtime.goexit

[2022-08-08T15:32:44.451Z]         	            		/usr/local/go/src/runtime/asm_amd64.s:1571

[2022-08-08T15:32:44.451Z]         	Test:       	TestExampleMaster
```

https://ci.pingcap.net/blue/organizations/jenkins/ut-check/detail/ut-check/28419/pipeline/

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

--- a/engine/pkg/meta/mock/clientconn_mock.go
+++ b/engine/pkg/meta/mock/clientconn_mock.go
@@ -42,6 +42,8 @@ func NewClientConnForSQLite(dsn string) (metaModel.ClientConn, error) {
 		return nil, err
 	}
 
+	db.SetMaxOpenConns(1)
+
 	return &sqliteClientConn{
 		db: db,
 	}, nil

